Fix VTMR: resolve conflicts with upstream

guokeno0 opened the pull request:
@alainjobart 

This is the same PR which removes the fix for streaming query RowsAffected

guokeno0 commented:
@erzel

guokeno0 commented:
Need someone to look at this, the problem will fail internal MR jobs on all single-sharded keyspaces

alainjobart commented:
Can we add a test somewhere for this? maybe tabletmanager.py would be a good place, create a new unit test there, you can setup the two tablets as master+rdonly in a single keyspace, and run vtgate.split_query on it (see custom_sharding.py for an example of how to do that). Thanks!

guokeno0 commented:
As discussed, I've changed SplitQuery to make decision by ShardingColumnType. I've also changed sandbox test to add ShardingColumnType to the fake topo.

alainjobart commented:
LGTM I'll work on some tests for this when submitted.

erzel commented on go/vt/vtgate/vtgate.go:
```diff
@@ -605,7 +605,10 @@ func (vtg *VTGate) SplitQuery(ctx context.Context, keyspace string, sql string,
-	if srvKeyspace.ShardingColumnName != "" {
+	// sharding_column_type != KeyspaceIdType_UNSET can happen in one of the following two cases:
+	// 1. We are querying a sharded keyspace;
+	// 2. We are querying an unsharded keyspace which is being sharded.
```
Thanks for the comment. It's clearer now.
